--- definitions.units	2017-11-28 20:17:13.021934545 +0100
+++ definitions.units.patched	2017-11-28 20:18:05.570776490 +0100
@@ -196,7 +196,9 @@
 US$       !      # The US dollar is chosen arbitrarily to be the primitive
                  #   unit of money.
 
-bit       !      # Basic unit of information (entropy).  The entropy in bits
+B         !
+byte      B
+bit       1|8 B  # Basic unit of information (entropy).  The entropy in bits
                  #   of a random variable over a finite alphabet is defined
                  #   to be the sum of -p(i)*log2(p(i)) over the alphabet where
                  #   p(i) is the probability that the random variable takes
@@ -246,12 +248,16 @@
 tebi-                   2^40     # Electrotechnical Commission aproved these
 pebi-                   2^50     # binary prefixes for use in 1998.  If you
 exbi-                   2^60     # want to refer to "megabytes" using the
+zebi-                   2^70
+yobi-                   2^80
 Ki-                     kibi     # binary definition, use these prefixes.
 Mi-                     mebi
 Gi-                     gibi
 Ti-                     tebi
 Pi-                     pebi
 Ei-                     exbi
+Zi-                     zebi
+Yi-                     yobi
 
 Y-                      yotta
 Z-                      zetta
@@ -267,6 +273,7 @@
 c-                      centi
 m-                      milli
 u-                      micro   # it should be a mu but u is easy to type
+mu-                     micro
 n-                      nano
 p-                      pico
 f-                      femto
@@ -623,6 +630,7 @@
 minute                  60 s
 min                     minute
 hour                    60 min
+h                       hour
 hr                      hour
 day                     24 hr
 d                       day
@@ -682,7 +690,7 @@
 pointangle              1|32 circle  # Used for reporting compass readings
 centrad                 0.01 radian  # Used for angular deviation of light
                                      # through a prism.
-mas                     milli arcsec # Used by astronomers
+mas                     milliarcsec  # Used by astronomers
 seclongitude            circle (seconds/day) # Astronomers measure longitude
                                      # (which they call right ascension) in
                                      # time units by dividing the equator into
@@ -788,8 +796,8 @@
 tempF(x) units=[1;K] domain=[-459.67,) range=[0,) \
                 (x+(-32)) degF + stdtemp ; (tempF+(-stdtemp))/degF + 32
 tempfahrenheit() tempF
-degfahrenheit           5|9 degC
-degF                    5|9 degC
+degfahrenheit           5|9 * degC
+degF                    5|9 * degC
 
 
 degreesrankine          degF              # The Rankine scale has the
@@ -801,7 +809,7 @@
 
 tempreaumur(x)    units=[1;K] domain=[-218.52,) range=[0,) \
                   x degreaumur+stdtemp ; (tempreaumur+(-stdtemp))/degreaumur
-degreaumur              10|8 degC # The Reaumur scale was used in Europe and
+degreaumur              10|8 * degC # The Reaumur scale was used in Europe and
                                   # particularly in France.  It is defined
                                   # to be 0 at the freezing point of water
                                   # and 80 at the boiling point.  Reaumur
@@ -873,8 +881,10 @@
 epsilon0                1/mu0 c^2        # permittivity of vacuum (exact)
 energy                  c^2              # convert mass to energy
 e                     1.6021766208e-19 C # electron charge
-h                   4.135667662e-15 eV s # Planck constant
-hbar                    h / 2 pi
+eV                      e V      # Energy acquired by a particle with charge e
+electronvolt            eV       #   when it is accelerated through 1 V
+hplanck             4.135667662e-15 eV s # Planck constant
+hbar                    hplanck / 2 pi
 spin                    hbar
 G               6.67408e-11 N m^2 / kg^2 # Newtonian gravitational constant
                                          #    This is the NIST 2006 value.
@@ -984,6 +994,13 @@
 H2O50C        0.98807 force gram / cm^3
 H2O100C       0.95838 force gram / cm^3
 
+# moved because ft is used before being defined (and otherwise parsed as femtotonne)
+inch                    2.54 cm
+in                      inch
+foot                    12 inch
+feet                    foot
+ft                      foot
+
 # Atomic constants
 
 Rinfinity            10973731.568539 /m  # The wavelengths of a spectral series
@@ -1050,12 +1067,12 @@
 # particle wavelengths: the compton wavelength of a particle is
 # defined as h / m c where m is the mass of the particle.
 
-electronwavelength      h / m_e c
+electronwavelength      hplanck / m_e c
 lambda_C                electronwavelength
-protonwavelength        h / m_p c
-lambda_C,p              protonwavelength
-neutronwavelength       h / m_n c
-lambda_C,n              neutronwavelength
+protonwavelength        hplanck / m_p c
+lambda_Cp               protonwavelength
+neutronwavelength       hplanck / m_n c
+lambda_Cn               neutronwavelength
 
 # Magnetic moments
 
@@ -1089,31 +1106,20 @@
 inHg                    inch Hg    
 inH2O                   inch water
 mmH2O                   mm water
-eV                      e V      # Energy acquired by a particle with charge e
-electronvolt            eV       #   when it is accelerated through 1 V
 lightyear               c julianyear # The 365.25 day year is specified in
 ly                      lightyear    # NIST publication 811
 lightsecond             c s
 lightminute             c min
-parsec                  au / tan(arcsec)    # Unit of length equal to distance
-pc                      parsec              #   from the sun to a point having
-                                            #   heliocentric parallax of 1
-                                            #   arcsec (derived from parallax
-                                            #   second).  A distant object with
-                                            #   paralax theta will be about
-                                            #   (arcsec/theta) parsecs from the
-                                            #   sun (using the approximation
-                                            #   that tan(theta) = theta).
-rydberg                 h c Rinfinity       # Rydberg energy
+rydberg                 hplanck c Rinfinity       # Rydberg energy
 crith                   0.089885 gram       # The crith is the mass of one
                                             #   liter of hydrogen at standard
                                             #   temperature and pressure.
 amagatvolume            molarvolume
 amagat                  mol/amagatvolume    # Used to measure gas densities
-lorentz                 bohrmagneton / h c  # Used to measure the extent
+lorentz                 bohrmagneton / hplanck c  # Used to measure the extent
                                             #   that the frequency of light
                                             #   is shifted by a magnetic field.
-cminv                   h c / cm            # Unit of energy used in infrared
+cminv                   hplanck c / cm            # Unit of energy used in infrared
 invcm                   cminv               #   spectroscopy.
 wavenumber              cminv
 kcal_mol                kcal_th / mol N_A   # kcal/mol is used as a unit of
@@ -1170,8 +1176,8 @@
 
 abampere                10 A            # Current which produces a force of
 abamp                   abampere        #   2 dyne/cm between two infinitely
-aA                      abampere        #   long wires that are 1 cm apart
-biot                    aA              # alternative name for abamp
+abA                     abampere        #   long wires that are 1 cm apart
+biot                    abA             # alternative name for abamp
 Bi                      biot
 abcoulomb               abamp sec
 abcoul                  abcoulomb
@@ -1393,8 +1399,8 @@
 
 # Reflected-light meter calibration constant with ISO 100 speed
 
-k1250                   12.5 (cd/m2) / lx s   # For Canon, Nikon, and Sekonic
-k1400                   14   (cd/m2) / lx s   # For Kenko (Minolta) and Pentax
+k1250                   12.5 (cd/m^2) / lx s   # For Canon, Nikon, and Sekonic
+k1400                   14   (cd/m^2) / lx s   # For Kenko (Minolta) and Pentax
 
 # Incident-light meter calibration constant with ISO 100 film
 
@@ -1448,8 +1454,8 @@
                                 # ASA PH2.12-1961, and ANSI PH2.7-1986
 #N_apex         0.3162 lx s     # value in ANSI PH2.7-1973
 N_exif          1|3.125 lx s    # value in Exif 2.3 (2010), making Sv(5) = 100
-K_apex1961      11.4 (cd/m2) / lx s    # value in ASA PH2.12-1961
-K_apex1971      12.5 (cd/m2) / lx s    # value in ANSI PH3.49-1971; more common
+K_apex1961      11.4 (cd/m^2) / lx s    # value in ASA PH2.12-1961
+K_apex1971      12.5 (cd/m^2) / lx s    # value in ANSI PH3.49-1971; more common
 C_apex1961      224 lx / lx s   # value in PH2.12-1961 (20.83 for I in 
                                 #   footcandles; flat sensor?)
 C_apex1971      322 lx / lx s   # mean value in PH3.49-1971 (30 +/- 5 for I in
@@ -1787,6 +1793,16 @@
 au                     astronomicalunit # ephemeris for the above described
                                         # astronomical unit.  (See the NASA
                                         # site listed above.)
+parsec                  au / 4.84813681113334e-06
+#parsec                 au / tan(arcsec)    # Unit of length equal to distance
+pc                      parsec              #   from the sun to a point having
+                                            #   heliocentric parallax of 1
+                                            #   arcsec (derived from parallax
+                                            #   second).  A distant object with
+                                            #   paralax theta will be about
+                                            #   (arcsec/theta) parsecs from the
+                                            #   sun (using the approximation
+                                            #   that tan(theta) = theta).
 solarmass                  1.9891e30 kg
 sunmass                       solarmass
 
@@ -2124,11 +2140,6 @@
 int                     3937|1200 ft/m   # Convert US Survey measures to
 int-                    int              #   international measures
 
-inch                    2.54 cm
-in                      inch
-foot                    12 inch
-feet                    foot
-ft                      foot
 yard                    3 ft
 yd                      yard
 mile                    5280 ft          # The mile was enlarged from 5000 ft
@@ -2276,6 +2287,7 @@
 # Liquid measure
 
 usgallon                231 in^3        # US liquid measure is derived from
+gallon                  usgallon
 gal                     gallon          # the British wine gallon of 1707.
 quart                   1|4 gallon      # See the "winegallon" entry below
 pint                    1|2 quart       # more historical information.
@@ -2768,7 +2780,7 @@
 
 number1can              10 usfloz
 number2can              19 usfloz
-number2.5can            3.5 uscups
+number2_5can            3.5 uscups
 number3can              4 uscups
 number5can              7 uscups
 number10can             105 usfloz
@@ -3198,9 +3210,9 @@
 
 btu                     btu_IT       # International Table BTU is the default
 britishthermalunit      btu
-btu_IT                  cal_IT lb degF / gram K
-btu_th                  cal_th lb degF / gram K
-btu_mean                cal_mean lb degF / gram K
+btu_IT                  cal_IT lb (degF) / gram K
+btu_th                  cal_th lb (degF) / gram K
+btu_mean                cal_mean lb (degF) / gram K
 quad                    quadrillion btu
 
 ECtherm                 1.05506e8 J    # Exact definition, close to 1e5 btu
@@ -3325,8 +3337,8 @@
                                        # of coal used in the 1950's
 barreloil               5.8 Mbtu       # Conventional value for barrel of crude
                                        # oil [E2].  Actual range is 5.6 - 6.3.
-naturalgas_HHV          1027 btu/ft3   # Energy content of natural gas.  HHV
-naturalgas_LHV          930 btu/ft3    # is for Higher Heating Value and
+naturalgas_HHV          1027 btu/ft^3  # Energy content of natural gas.  HHV
+naturalgas_LHV          930 btu/ft^3   # is for Higher Heating Value and
 naturalgas              naturalgas_HHV # includes energy from condensation
                                        # combustion products.  LHV is for Lower
                                        # Heating Value and excludes these.
@@ -3361,7 +3373,7 @@
 
 # Celsius heat unit: energy to raise a pound of water 1 degC
 
-celsiusheatunit         cal lb degC / gram K
+celsiusheatunit         cal lb (degC) / gram K
 chu                     celsiusheatunit
 
 POWER                   watt
@@ -3421,13 +3433,13 @@
 Uvalue                  1/Rvalue
 europeanUvalue          watt / m^2 K
 RSI                     degC m^2 / W
-clo                     0.155 degC m^2 / W # Supposed to be the insulance
+clo                     0.155 * degC m^2 / W # Supposed to be the insulance
                                            # required to keep a resting person
                                            # comfortable indoors.  The value
                                            # given is from NIST and the CRC,
                                            # but [5] gives a slightly different
                                            # value of 0.875 ft^2 degF hr / btu.
-tog                     0.1 degC m^2 / W   # Also used for clothing.
+tog                     0.1 * degC m^2 / W   # Also used for clothing.
 
 
 # The bel was defined by engineers of Bell Laboratories to describe the
@@ -3801,7 +3813,7 @@
 europeanpoint           didotpoint        # for the point used in Europe
 cicero                  12 didotpoint
 
-stick                   2 inches
+stick                   2 in
 
 # Type sizes
 
@@ -3858,8 +3870,8 @@
 
 INFORMATION             bit
 
-nat                     (1/ln(2)) bits       # Entropy measured base e
-hartley                 log2(10) bits        # Entropy of a uniformly
+nat                     (1/0.693147180559945) bits       # Entropy measured base e
+hartley                 3.321928094887363 bits        # Entropy of a uniformly
 ban                     hartley              #   distributed random variable
                                              #   over 10 symbols.
 dit                     hartley              # from Decimal digIT
@@ -3874,8 +3886,8 @@
                                              #   to symbols per second.  Modern
                                              #   modems transmit several bits
                                              #   per symbol.
-byte                    8 bit                # Not all machines had 8 bit
-B                       byte                 #   bytes, but these days most of
+#byte                    8 bit               # Not all machines had 8 bit
+#B                       byte                #   bytes, but these days most of
                                              #   them do.  But beware: for
                                              #   transmission over modems, a
                                              #   few extra bits are used so
@@ -4230,7 +4242,7 @@
 boardfeet               boardfoot
 fbm                     boardfoot    # feet board measure
 stack                   4 yard^3     # British, used for firewood and coal [18]
-rick                    4 ft 8 ft 16 inches # Stack of firewood, supposedly
+rick                    4 ft 8 ft 16 in # Stack of firewood, supposedly
                                      #   sometimes called a face cord, but this
                                      #   value is equal to 1|3 cord.  Name
                                      #   comes from an old Norse word for a
@@ -5310,7 +5322,7 @@
 
 pa                      Pa
 ev                      eV
-hg                      Hg
+#hg                     Hg
 oe                      Oe
 mh                      mH
 rd                      rod
@@ -5543,7 +5555,8 @@
 Mag                     Maz gravity        # force
 Maz                     Volm kg / oldliter # mass based on water
 
-Tm                      Tim                # Abbreviations
+#gross conflict with terameter
+#Tm                      Tim                # Abbreviations
 Gf                      Grafut
 Sf                      Surf
 Vm                      Volm
@@ -6294,7 +6307,6 @@
 ton                     uston
 scruple                 apscruple
 fluidounce              usfluidounce
-gallon                  usgallon
 bushel                  usbushel
 quarter                 quarterweight
 cup                     uscup
@@ -6380,7 +6392,7 @@
 ℈                       scruple
 ℥                       ounce
 ℔                       lb
-ℎ                       h
+ℎ                       hplanck
 ℏ                       hbar
 ‰                       1|1000
 ‱                       1|10000
